onnxruntime/core/optimizer/embed_layer_norm_fusion.cc

```diff
@@ -13,6 +14,10 @@ namespace onnxruntime {
 
 // Add a Cast to convert Input from int64 to int32.
 static NodeArg* CastToInt32(Graph& graph, NodeArg* input, ProviderType provider_type) {
+  auto data_type = input->TypeAsProto()->tensor_type().elem_type();
+  if (data_type != ONNX_NAMESPACE::TensorProto_DataType_INT64) {
```

Comment:
Change condition to  (data_type == ONNX_NAMESPACE::TensorProto_DataType_INT32). Otherwise, the logic depending on that data type is either int64 or int32. For example, some day when we allow int16.